### DISCUSSION TITLE
[Signal Store] Get state and actions of another store without customFeature

### DISCUSSION THREAD

**Discussion by BDominik** [ngrx/platform#4609] [Q&A] [answered]
Hello. 
I want to use a signal store as component store.
So I have a **home-page.store.ts**. 
And I also have a ### products-page.store.ts
And I want to have a third **providedIn: 'root'** store like **products.entity.store**.
I've tried to create my **products.entity.stor**e as a custom feature.
But then I have it two times under the home and products page store.
How can I have two component store **home-page** and **products.page** and to connect it to a providedIn root separate component. 
Without merging it in the component or in a separate view models.
Thanks.

**Comment by BDominik** [comment, marked as answer]
I can see that I could just inject my entity store into the home page store and use it like that, but I'm not sure if it's a good idea.
![image](https://github.com/user-attachments/assets/df4c8897-d358-449a-8c15-17ca5a0af909)

**Reply by rainerhahnekamp** [reply]
Hey, you have your answer. Why do you think that this is the wrong approach?

**Reply by BDominik** [reply]
Tbh because of the custom feature I thought maybe there is also a similar way to import / connect another store. 
Thanks for the answer. And also thanks for your Youtube videos. I learned a lot from you.

**Reply by rainerhahnekamp** [reply]
Ah, now I understand. There’s been a noticeable focus on questions about the relationship between two stores, and I wasn’t sure why. It makes sense now—one might assume an extension is needed in this case.

But as before, simply use inject and treat the other SignalStore like any other service.

(Happy to hear you enjoyed my videos!)

**Reply by BDominik** [reply]
So here is what I did in the end.
I use signal store as a component store. But I want to get data from multiple global store, where I have my entities.
So I created a feature store named withCryptoData and this store is injecting the global stores and combines the data together.
Because most of my component stores will need those data plus the actual component logic.
And then I can just use this withCryptoData there and access the computed values and the methods.
I really like this way.
I have one more question regarding the treeshakability.
So in one of the videos you said it's better to use functions.
Right now I just created a service, where I'm building the data structure for the view. Like portfolio list (which uses the crypto-details, the transactions and the crypto price entities), and providing it for the withCryptoData future store.
So would it make more sense to have a file with functions only instead a service?

Thanks for the answer upfront.

**Reply by BDominik** [reply]
![image](https://github.com/user-attachments/assets/e06a1165-2ff2-4ab8-8bf1-0318dc77793d)
This is how it's looks like.
